### PR TITLE
Add section on supporting the range of devices and platforms.

### LINKED ISSUE #173: add principle on media independence (HTML Design Principle 5.1)
The HTML design principles has a section [5.1. Media Independence](https://www.w3.org/TR/html-design-principles/#media-independence).  @fantasai's [blog post](http://fantasai.inkedblade.net/weblog/2019/designing-css/#xp) also has some material that could be incorporated here (following up from https://github.com/w3ctag/design-principles/issues/126#issuecomment-610117172).

### PR DESCRIPTION
**Pull request by dbaron**
I wrote this in a breakout with @plinss during our virtual face-to-face.

Fixes #173.

***
<a href="https://pr-preview.s3.amazonaws.com/dbaron/design-principles/pull/198.html#devices-platforms" title="Last updated on May 29, 2020, 2:22 AM UTC (00c2aa3)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3ctag/design-principles/198/049ce03...dbaron:00c2aa3.html" title="Last updated on May 29, 2020, 2:22 AM UTC (00c2aa3)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -180,6 +180,30 @@ by showing the user that certain things won't happen without their permission.
 But frequently asking users for consent can also show how scary a place the web is
 by showing how many sites are willing to ask for intrusive and unnecessary permissions.
 
+<h3 id="devices-platforms">Support the full range of devices and platforms</h3>
+
```
Comment (hober) [inline index.bs:184]: Please add a `<!-- was "Media Independence" in the HTML Design Principles -→` comment here. I've been doing this for principles which get ported over. (See for instance #190.)

Comment (dbaron) [reply index.bs:185]: Done in 717bc8d.

File: index.bs [hunk 2]
```diff
@@ -180,6 +180,31 @@ by showing the user that certain things won't happen without their permission.
...
+<h3 id="devices-platforms">Support the full range of devices and platforms</h3>
+<!-- was "Media Independence" in the HTML Design Principles -->
+
+Features on the web should work across different input and output devices,
+different platforms, and different media,
+as much as this is reasonably possible.
+For example, they should not be specific to a particular screen size,
+a particular operating system or browser engine,
+or assume the user is using a mouse.
+Some features don't reasonably work across all of these cases:
```
Comment (alice) [inline index.bs:192]:
```suggestion

It is not always possible for a feature to work in all contexts:
```

File: index.bs [hunk 3]
```diff
@@ -180,6 +180,31 @@ by showing the user that certain things won't happen without their permission.
...
+for example, hyperlinks don't work when printed on paper,
+and '':hover'' doesn't work on touch input devices where there isn't a mouse pointer position,
```
Comment (alice) [inline index.bs:194]:
```suggestion
and '':hover'' doesn't work on touch input devices where there isn't a mouse pointer position.
```

File: index.bs [hunk 4]
```diff
@@ -180,6 +180,31 @@ by showing the user that certain things won't happen without their permission.
...
+but these features are still valuable across a range of devices,
```
Comment (alice) [inline index.bs:195]:
```suggestion
However, these features are still valuable across a range of devices,
```

File: index.bs [hunk 5]
```diff
@@ -180,6 +180,31 @@ by showing the user that certain things won't happen without their permission.
...
+and could be adapted to devices that don't support their original intent
```
Comment (alice) [inline index.bs:196]:
```suggestion
and can be adapted to devices that don't support their original intent
```

File: index.bs [hunk 6]
```diff
@@ -180,6 +180,31 @@ by showing the user that certain things won't happen without their permission.
...
+(for example, hyperlink targets could be converted to footnotes when printing).
```
Comment (alice) [inline index.bs:197]:
```suggestion
(for example, hyperlink targets may be converted to footnotes when printing).
```

File: index.bs [hunk 7]
```diff
@@ -180,6 +180,31 @@ by showing the user that certain things won't happen without their permission.
...
+
+Features should also be designed so that the easiest way to use them
+works across devices as well.
+For example, the simplest way of using a feature for positioning of content
```
Comment (alice) [inline index.bs:201]: Could this be a specific, real-world example?

Comment (dbaron) [reply index.bs]: Converted to one in 00c2aa3.

**Pull request merged**